numpy/_array_api/_array_object.py

```diff
@@ -150,7 +150,7 @@ def __abs__(self: array, /) -> array:
         return self.__class__._new(res)
 
     @np.errstate(all='ignore')
-    def __add__(self: array, other: array, /) -> array:
+    def __add__(self: array, other: Union[int, float, array], /) -> array:
```

Comment:
```suggestion
    def __add__(self: array, other: Union[float, array], /) -> array:
```
[PEP 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower) prescribes that `int` should be treated as a sub-type of `float`, so `float` is equivalent to `Union[int, float]`. 

You could of course keep things as they are if you'd prefer being very explicit, but note that it serves no other practical functionality.

Reply:
Yes, I think I pointed that out somewhere before; it's arguable whether it has value as documentation or not.

Reply:
There are some array api functions that only work with floating point dtype (not int), but that fortunately doesn't apply to any of the operators (although I'm still confused by `__pow__` and `__truediv__`, since the spec says the functions `pow` and `divide` only need to accept floating point dtypes).

Reply:
At https://github.com/data-apis/array-api/pull/221 and here I've updated `__pow__` and `__truediv__` to not accept integer dtypes (or `int`). It's unfortunate that the PEP makes it impossible to actually distinguish the two, but I think we should write the type signatures in a way that does distinguish them because of methods like these.